current_symbols.py

```diff
@@ -1,31 +1,72 @@
 import pandas as pd
-import sys
-from datetime import date
-
-def evaLink(elem1, elem2, predicate):
-    return '(EvaluationLink \n' + '\t(PredicateNode "'+predicate+'")\n' + '\t(ListLink \n'+ '\t\t'+ elem1 + '\n\t\t'+ elem2 + '))\n'
-
-def toAtomese(data):
-    df =  pd.read_csv(data, sep='\t')
-    df = df.dropna()
-    print("Started importing ")
-    with open("dataset/current_symbols_{}.scm".format(str(date.today())), "w") as f:
-        for i in range(len(df)):
-            prev = df.iloc[i]['Previous symbols']
-            current = df.iloc[i]['Approved symbol'].strip().upper()
-            name = df.iloc[i]['Approved name']
-            if prev:
-                for p in prev.split(","):
-                    f.write(evaLink('(GeneNode "{}")'.format(p.strip().upper()), '(GeneNode "{}")'.format(current), "has_current_symbol"))
-                f.write(evaLink('(GeneNode "{}")'.format(current), '(ConceptNode "{}")'.format(name), "has_name"))
-    print("Done")
-    
-if __name__ == "__main__":
-    # Source https://www.genenames.org/download/custom/ 
-    # - Select columns Approved symbol, Approved name and Previous symbol 
-    # - save into file, and give as an input
-    try:
-        data = sys.argv[1]
-    except:
-        data = "raw_data/custom_current.txt" 
-    toAtomese(data)
+import requests
+import io
+import numpy as np
+
+# Splits string entry to rows and expand the dataframe
+def explode(df, lst_cols, fill_value='NA', preserve_index=False):
+    # make sure `lst_cols` is list-alike
+    if (lst_cols is not None
+        and len(lst_cols) > 0
+        and not isinstance(lst_cols, (list, tuple, np.ndarray, pd.Series))):
+        lst_cols = [lst_cols]
+    # all columns except `lst_cols`
+    idx_cols = df.columns.difference(lst_cols)
+    # calculate lengths of lists
+    lens = df[lst_cols[0]].str.len()
+    # preserve original index values    
+    idx = np.repeat(df.index.values, lens)
+    # create "exploded" DF
+    res = (pd.DataFrame({
+                col:np.repeat(df[col].values, lens)
+                for col in idx_cols},
+                index=idx)
+             .assign(**{col:np.concatenate(df.loc[lens>0, col].values)
+                            for col in lst_cols}))
+    # append those rows that have empty lists
+    if (lens == 0).any():
+        # at least one list in cells is empty
+        res = (res.append(df.loc[lens==0, idx_cols], sort=False)
+                  .fillna(fill_value))
+    # revert the original index order
```

Comment:
Does sorting revert the original index order?

Reply:
Sort is set to False, but since we add additional rows we need to sort the index back to its original order.